packages/blocks/src/image-block/image-block.ts

```diff
@@ -158,6 +159,14 @@ export class ImageBlockComponent extends CaptionedBlockComponent<
     `;
   }
 
+  toZIndex() {
+    return this.rootService?.layer.getZIndex(this.model) ?? 1;
+  }
+
+  updateZIndex() {
```

Comment:
It looks suboptimal that we need to hard-code the same default methods for these block types. Inheriting from `GfxBlockComponent` avoids the hardcoding, but seems to be conflict with `CaptionedBlockComponent`.

Do you think we need a better block-element relationship modeling (probably not in this PR) to avoid this @Saul-Mirone?

Reply:
This is a temporary workaround. Because we need to render both the page mode image and the edgeless mode image at the same time, the image block cannot extend the GfxBlockComponent. I am adding those methods manually to make it renderable in edgeless mode.

We need to support defining different views for the same type of block depending on its parent type. In this case, an image in the note block should use the "page image block," while an image in the surface block should use the "edgeless image block."